### ISSUE TITLE
Add network community detection util

### ISSUE THREAD

**Issue by EnricoGavagnin** [nestauk/nesta_ds_utils#48]
This will include a consensus_clustering util to determine a stable clustering of data by using a 3-step process.

**Comment by emily-bicks** [comment]
Shall we break this into several issues? Maybe one issue for each of the features we have discussed?
1. Generate co-occurence network based on lists of entities
2. Generate knn network
3. Run community detection on a network

**Comment by EnricoGavagnin** [comment]
Yes, sorry. This was supposed to be the issue for the community detection util, but I realised it's not very clear. I will rename this and open two new issues for the other two features.
